shared/src/ledger/storage/mockdb.rs

```diff
@@ -289,11 +289,59 @@ impl DB for MockDB {
         }
     }
 
+    fn read_merkle_tree_stores(
+        &self,
+        height: BlockHeight,
+    ) -> Result<Option<MerkleTreeStoresRead>> {
+        let mut merkle_tree_stores = MerkleTreeStoresRead::default();
+        let height_key = Key::from(height.to_db_key());
+        let tree_key = height_key
+            .push(&"tree".to_owned())
+            .map_err(Error::KeyError)?;
+        for st in StoreType::iter() {
+            let prefix_key =
+                tree_key.push(&st.to_string()).map_err(Error::KeyError)?;
+            let root_key = prefix_key
+                .push(&"root".to_owned())
+                .map_err(Error::KeyError)?;
+            let bytes = self.0.borrow().get(&root_key.to_string()).cloned();
+            match bytes {
+                Some(b) => {
+                    let root = types::decode(b).map_err(Error::CodingError)?;
+                    merkle_tree_stores.set_root(st, root);
+                }
+                None => return Ok(None),
+            }
+
+            let store_key = prefix_key
+                .push(&"store".to_owned())
+                .map_err(Error::KeyError)?;
+            let bytes = self.0.borrow().get(&store_key.to_string()).cloned();
+            match bytes {
+                Some(b) => {
+                    let store = types::decode(b).map_err(Error::CodingError)?;
+                    merkle_tree_stores.set_store(st, store);
+                }
+                None => return Ok(None),
+            }
+        }
+        Ok(Some(merkle_tree_stores))
+    }
+
     fn read_subspace_val(&self, key: &Key) -> Result<Option<Vec<u8>>> {
         let key = Key::parse("subspace").map_err(Error::KeyError)?.join(key);
         Ok(self.0.borrow().get(&key.to_string()).cloned())
     }
 
+    fn read_subspace_val_with_height(
+        &self,
+        key: &Key,
+        _height: BlockHeight,
+    ) -> Result<Option<Vec<u8>>> {
+        // read the value at the latest height for the mock DB
+        self.read_subspace_val(key)
```

Comment:
I feel a bit hesitant about ignoring the height as this diverges from the rockdb behavior. I’d be tempted to just use `unimplemented!()` here

Reply:
Thank you for the comment! make sense! I've replaced it with `unimplemented!()`